### ISSUE TITLE
Error calling JSON in IE with jQuery

### ISSUE THREAD

**Issue by gottherim4** [almende/vis#1586]
Hello, I am very new to javascript development and have run into a problem I believe could be easy to resolve. My issue is only in IE 11 (haven't tested older IE versions) and my script/html works on Chrome and Firefox. I am seeking help here at vis.js because in the IE 11 console, there are multiple error messages that refer to vis.js (version 4.11) line 26441: 
dFill = 'M' + pathArray[0][0] + "," + pathArray[0][1] + " " + this.serializePath(pathArray, type, false) + ' V' + zero + ' H' + pathArray[0][0] + " Z";

and the error from IE: SCRIPT5007: Unable to get property '0' of undefined or null reference

I am making an ajax call with jQuery to a JSON file, then creating several DataSets for a 2Dgraph from the JSON data. In IE the 2D graphs are being drawn, and the JSON data (JSON-lint verified) is being read. I know the data is being accessed by IE because the 2Dgraph legends pull the object names like the other browsers do, but there is just no data being shown on the graph. I have found several suggestions online to change the contentType, dataType, etc. in the ajax call but none of them have made a difference. 

Any ideas on the cause of my issue? Thank you ahead of time from a vis.js rookie.

**Comment by AlexDM0** [comment]
Hi,

It would be a great help if we could reproduce this. If you save the data locally and load it into graph2d, do you get the same error? It sounds like there are illegal characters or a datapoint that has an x but no y or something. It is valid JSON but not valid vis data. Can you post the data here?

Regards

**Comment by ludost** [comment]
This might have been fixed in the development branch, I've found and fixed a case where the path is not yet set at initial graph load.

**Comment by gottherim4** [comment]
I believe I found a solution, and I do not believe vis.js was to blame. Alex, I did what you suggested and local data array still did not graph data points in IE.

The line 26441 I mentioned above was creating shading effects when drawing data points. I turned off all shading, lost the error messages, but still did not see data in IE. After trial and error, I found the date/time field format I was using (YYYY MM DD HH:MM:SS) was not acceptable to IE (or Safari on mobile). After changing the date field format to (YYYY-MM-DD HH:MM:SS) I was able to see data points in IE and Safari. I turned shading back on and no error messages. 

Thank you gentlemen for getting me thinking and all your work on this project.
